internal/server/server.go

```diff
@@ -152,12 +161,17 @@ func ToClusterTrial(t *redskyv1beta1.Trial, suggestion *redskyapi.TrialAssignmen
 	}
 
 	for _, a := range suggestion.Assignments {
-		if v, err := a.Value.Int64(); err == nil {
-			t.Spec.Assignments = append(t.Spec.Assignments, redskyv1beta1.Assignment{
-				Name:  a.ParameterName,
-				Value: v,
-			})
+		var v intstr.IntOrString
+		if a.Value.IsString {
+			v = intstr.FromString(a.Value.StrVal)
+		} else {
+			v = intstr.FromInt(int(a.Value.Int64Value()))
```

Comment:
I went back and forth on this. At the end of the day the bounds we collect in the Kube API are strongly typed to `int32` so if the server honors the configured bounds then we will never overflow. Everything in `suggest` is done with `int64` math because we do not technically have the protection of what is in the Kube API when dealing directly with the ML backend (in fact we still deal with floating point numbers there). The `FromInt` call will report overflow on 64-bit architectures to the log.

I will update this, but because we aren't reporting failures, the behavior will likely be to just cap values to what fits in a 32-bit integer: again the assumption being that the code would be effectively unreachable.